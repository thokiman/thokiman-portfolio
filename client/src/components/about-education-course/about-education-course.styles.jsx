import { Link } from "react-router-dom";
import styled, { css, keyframes } from "styled-components";

export const AboutEducationCourseContent = styled.div`
  padding: 5px;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  width: 92%;
  margin: 0 25px;
  position: relative;
  top: 10px;
`;

export const AboutEducationCourseTitle = styled.div`
  display: inline-block;
  width: 90%;
  font-size: 500%;
  text-align: left;
  line-height: 1.3em;
  letter-spacing: 0.04em;
  font-kerning: auto;
  padding: 0 2px;
  color: #e3e1e4;
  @media only screen and (max-width: 800px) {
    font-size: 18vw;
  }
`;

export const AboutEducationCourseAttributes = styled.div`
  display: grid;
  grid-auto-columns: repeat(4, 1fr);
  grid-auto-rows: repeat(3, 1fr);
  gap: 2px 0px;
  grid-template-areas:
    "education-experience-company education-experience-company education-experience-organization education-experience-organization"
    "education-experience-company education-experience-company education-experience-date education-experience-date"
    "education-experience-company education-experience-company education-experience-link education-experience-link";
  width: 100%;
  height: 100%;
`;

const AboutEducationCourseStyles = css`
  text-align: center;
  line-height: 1.3em;
  letter-spacing: 0.04em;
  font-kerning: auto;
  border-bottom: #74c6d4 solid 2px;
  color: #e3e1e4;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const AboutEducationCourseName = styled.div`
  ${AboutEducationCourseStyles}
  grid-area: education-experience-company;
  font-size: 200%;
  padding: 10px 0;
  width: 95%;
  @media only screen and (max-width: 800px) {
    text-align: left;
    font-size: 6vw;
  }
`;

export const AboutEducationCourseOrganization = styled.div`
  ${AboutEducationCourseStyles}
  grid-area: education-experience-organization;
  font-size: 150%;
  padding: 0 10px;
  @media only screen and (max-width: 800px) {
    padding: 4px;
    font-size: 4.5vw;
  }
`;

export const AboutEducationCourseDate = styled.div`
  ${AboutEducationCourseStyles}
  grid-area: education-experience-date;
  padding: 10px 0;
  font-size: 150%;
  @media only screen and (max-width: 800px) {
    font-size: 4.5vw;
    padding: 4px;
  }
`;

export const AboutEducationCourseLink = styled.div`
  grid-area: education-experience-link;
  margin: 10px 0;
  display: flex;
  align-items: center;
  justify-content: center;
`;
const aboutCourseLink = keyframes`
                    from {
                        opacity: 0.3;

                    }

                    to {
                        opacity: 1;

                    }

`;
export const AboutCourseLink = styled(Link)`
  ${AboutEducationCourseStyles}
  width: 100%;
  height: 100%;
  padding: 4px;
  text-decoration: none;
  font-size: 120%;
  cursor: pointer;
  @media only screen and (max-width: 800px) {
    font-size: 4.5vw;
  }

  &:hover {
    border-bottom: #3f4249 solid 2px;
    display: flex;
    align-items: center;
    justify-content: center;
    animation: ${aboutCourseLink} 0.5s ease-in-out;
  }
`;

AboutEducationCourseContent.displayName = "AboutEducationCourseContent";
AboutEducationCourseTitle.displayName = "AboutEducationCourseTitle";
AboutEducationCourseAttributes.displayName = "AboutEducationCourseAttributes";
AboutEducationCourseName.displayName = "AboutEducationCourseName";
AboutEducationCourseOrganization.displayName =
  "AboutEducationCourseOrganization";
AboutEducationCourseDate.displayName = "AboutEducationCourseDate";
AboutEducationCourseLink.displayName = "AboutEducationCourseLink";
AboutCourseLink.displayName = "AboutCourseLink";
