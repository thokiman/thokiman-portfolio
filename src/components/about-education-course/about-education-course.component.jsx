import React from "react";

import EducationGrid from "../education-grid/education-grid.component";
import {
  AboutCourseLink,
  AboutEducationCourseAttributes,
  AboutEducationCourseContent,
  AboutEducationCourseDate,
  AboutEducationCourseLink,
  AboutEducationCourseName,
  AboutEducationCourseOrganization,
  AboutEducationCourseTitle,
} from "./about-education-course.styles";

const AboutEducationCourse = ({ lastCourse }) => {
  return (
    <AboutEducationCourseContent>
      <AboutEducationCourseTitle>Course</AboutEducationCourseTitle>
      {lastCourse.map(
        ({ id, name, organization, field, date, credentialUrl, imagePath }) => {
          return (
            <div key={id}>
              <AboutEducationCourseAttributes>
                <AboutEducationCourseName>{name}</AboutEducationCourseName>
                <AboutEducationCourseOrganization>
                  {organization}
                </AboutEducationCourseOrganization>
                <AboutEducationCourseDate>{date}</AboutEducationCourseDate>
                <AboutEducationCourseLink>
                  <AboutCourseLink
                    target="_blank"
                    to={{ pathname: credentialUrl }}
                  >
                    Professional Certificate - {field.charAt(0).toUpperCase()}
                    {field.slice(1)}
                  </AboutCourseLink>
                </AboutEducationCourseLink>
              </AboutEducationCourseAttributes>
              <EducationGrid imagePath={imagePath} />
            </div>
          );
        }
      )}
    </AboutEducationCourseContent>
  );
};

export default AboutEducationCourse;
