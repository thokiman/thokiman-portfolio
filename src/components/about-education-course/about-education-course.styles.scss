.education-course-content {

    padding: 5px;
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
    width: 92%;

    margin: 0 25px;
    position: relative;
    top: 10px;

    .education-course-title {

        display: inline-block;
        width: 90%;
        font-size: 500%;
        text-align: left;
        line-height: 1.3em;
        letter-spacing: 0.04em;
        font-kerning: auto;
        padding: 0 2px;
        color: #e3e1e4
    }

    .education-course-attributes {


        display: grid;
        display: grid;
        grid-auto-columns: repeat(4, 1fr);
        grid-auto-rows: repeat(3, 1fr);
        gap: 2px 0px;
        grid-template-areas:
            "education-experience-company education-experience-company education-experience-organization education-experience-organization"
            "education-experience-company education-experience-company education-experience-date education-experience-date"
            "education-experience-company education-experience-company education-experience-link education-experience-link";
        width: 100%;
        height: 100%;


        .education-course-name {
            grid-area: education-experience-company;
            padding: 10px 0;
            font-size: 200%;
            text-align: center;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;
            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;

            width: 95%;
        }

        .education-course-organization {

            grid-area: education-experience-organization;
            font-size: 150%;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;
            text-align: center;
            padding: 0 10px;

            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;



        }

        .education-course-date {

            grid-area: education-experience-date;
            padding: 10px 0;
            font-size: 150%;
            text-align: center;
            line-height: 1.3em;
            letter-spacing: 0.04em;
            font-kerning: auto;

            border-bottom: #74c6d4 solid 2px;
            color: #e3e1e4;
            display: flex;
            align-items: center;
            justify-content: center;




        }

        .education-course-link {
            grid-area: education-experience-link;
            margin: 10px 0;

            display: flex;
            align-items: center;
            justify-content: center;



            .course-link {
                width: 100%;
                height: 100%;
                padding: 4px;
                text-decoration: none;
                font-size: 120%;
                text-align: center;
                line-height: 1.3em;
                letter-spacing: 0.04em;
                font-kerning: auto;
                color: #e3e1e4;
                cursor: pointer;
                display: flex;
                align-items: center;
                justify-content: center;

                &:hover {

                    border-bottom: #3f4249 solid 2px;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    animation: course-link 0.5s ease-in-out;
                }

                @keyframes course-link {
                    from {
                        opacity: 0.3;

                    }

                    to {
                        opacity: 1;

                    }
                }
            }






        }



    }

    .education-course-images {}
}