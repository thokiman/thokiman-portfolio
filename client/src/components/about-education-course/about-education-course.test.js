import React from "react";
import { shallow } from "enzyme";

import { AboutEducationCourse } from "./about-education-course.component";

describe("AboutEducationCourse component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      lastCourse: mockArray,
    };
    wrapper = shallow(<AboutEducationCourse {...mockProps} />);
  });

  it("should render AboutEducationCourse component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
