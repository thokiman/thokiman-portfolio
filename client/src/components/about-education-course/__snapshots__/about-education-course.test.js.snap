// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutEducationCourse component should render AboutEducationCourse component 1`] = `ShallowWrapper {}`;
